### DISCUSSION TITLE
Livewire V3 | Non blocking action request.

### DISCUSSION THREAD

**Discussion by nanodreams** [livewire/livewire#7081] [1. Help]
Hi, 

I'm currently implementing a UI where users can interact and trigger different actions. While some of the actions might take q bit of time ~30 seconds, seems livewire is "chaining" all the trigger actions and excuting them in "sequencial" order once the first is finished, then it will trigger the next.

Is there anyway we can "trigger" several actions in parallel? Like if the users click one action then 5 seconds after, if the user triggers a second one, I would like to have another trigger event to the backend.

RIght now, if I trigger two actions within 5 seconds, the second action won't be triggered after the first one is finished.

Thanks for your help in advance.

PS: I thought something related to the defer, live ... but those seems more for "binding" to the state, rather than action / events dispatching.

**Comment by eilerth** [comment]
I found this to be a major issue with upgrading my project to livewire v3 and submitted a discussion post [here](https://github.com/livewire/livewire/discussions/6875), but ultimately submitted this PR to resolve the problem: https://github.com/livewire/livewire/pull/7102

Essentially, it allows you to add "isolate" to a livewire component to prevent it from bundling it's requests with other livewire components.
`<livewire:my-component isolate />`

**Reply by nanodreams** [reply]
Thanks @eilerth  for taking the time for answering and more important ,.. the PR. 
 Just in case someone ended up here, the PR from @eilerth  higlights the need and I also wanted to mention that there might be initially "two levels of isolation" (from my initial point of view).
 
 Scenario A (yours): Configure a component to act like V2 (totally isolated)
`<livewire:my-component isolate />`

Scenario B: Where you can control each action/event how to be triggered, independed of if they are within the same component or not.
```
<button wire:click.isolate="myAction" />
<button @click="$dipatch('myAction').isolate()"/>
$this->dispatch('myAction')->self()->isolated();
```

**Reply by eilerth** [reply]
@nanodreams, I wonder if those asynchronous actions would make sense to be split into their own components? One of my thoughts is that each response for the livewire component may have new html to render for the component, and if there are multiple requests from the same component running in parallel, the response from one action's request could overwrite the html from another. 

Example flow:
- You click action "A"
- While action "A" is running, you click action "B"
- Action "B" gets a response back and rewrites html for the component to reflect changes made by that action
- Action "A" then gets a response back and rewrites html for the component, but since it was triggered before action "B", it wasn't aware of what action "B" was doing, so it overwrites what action "B" did entirely.

Perhaps what I'm describing has nothing to do with your use case. Sorry if that's getting too deep. It's also possible I might not be thinking about it quite right! 

I can see another use case that might align better with yours, where there is one component that is handling these actions, and other components that are listening and updating independently. 

In any case, I do see how having granular control over each action getting isolated could be helpful.
